File: BEPiD/BepidAulas/aula23/Aula23Storyboard/Aula23Storyboard/ViewController.m
//
//  ViewController.m
//  Aula23Storyboard
//
//  Created by Danilo Barros Mendes on 5/11/15.
//  Copyright (c) 2015 Danilo Barros Mendes. All rights reserved.
//

#import "ViewController.h"
#import "DetailsViewController.h"

@interface ViewController ()

@property (strong, nonatomic) NSMutableArray *numbers;

@property (nonatomic) NSString *string;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Minha Primeira TV";
    
    self.numbers = [[NSMutableArray alloc]  initWithObjects:@"Um", @"Dois", @"Tres",
                    @"Quatro", @"Cinco", @"Seis",
                    @"Sete", @"Oito", @"Nove",
                    @"Dez", nil];
    
    
}

-(NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{
    return 1;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return self.numbers.count;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    static NSString *cellIdentifier = @"Cell";
    
    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault
                                                   reuseIdentifier:cellIdentifier];
    
    cell.textLabel.text = [self.numbers objectAtIndex:indexPath.row];
    
    return cell;
    
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    
    _string = [self.numbers objectAtIndex:indexPath.row];
    
    [self performSegueWithIdentifier:@"details" sender:self];

}

-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    
    if ([segue.identifier isEqualToString:@"details"]) {
        
        DetailsViewController *detailsView = segue.destinationViewController;
        
        [detailsView setDetails:_string];
        NSLog(@"oioi");
    }else{
        DetailsViewController *detailsView = segue.destinationViewController;
        
        [detailsView setDetails:@"Null"];
        
        NSLog(@"eiei");
    }
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: BEPiD/BepidAulas/aula27/CoreDataStarter/CoreDataDemo/ViewController.swift
//
//  ViewController.swift
//  CoreDataDemo
//
//  Created by Gustavo Salles on 30/05/15.
//  Copyright (c) 2015 Gustavo Salles. All rights reserved.
//

import UIKit


class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    @IBOutlet weak var tableInstituicoes: UITableView!
    
    var minStudents: Int = 0
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "BEPiD's"
        tableInstituicoes.registerClass(UITableViewCell.self, forCellReuseIdentifier: "Cell")

        self.tableInstituicoes.reloadData()
    }
    
    
    override func viewWillAppear(animated: Bool) {
        self.tableInstituicoes.reloadData()
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return BepidModel.sharedInstance().allBepids!.count
    }

    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {

        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath) as! UITableViewCell
        
        cell.textLabel?.text = "Teste"
        
        return cell
    }

    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        
        self.performSegueWithIdentifier("showBepidDetail", sender: self)
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "showBepidDetail" {
            let bvc = segue.destinationViewController as! BepidDetail
            
        }
    }

    @IBAction func filtraInstituicao(sender: AnyObject) {
        
    }

    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

